### PR TITLE
Support for thin type file structure

### PR DESCRIPTION
**Pull request by oven8**
I've added modifications to support thin type file structures. The main differences are noted below:

1. There are 312 words instead of 273.
2. Particle data and Cherenkov data contains 8 words (instead of 7) for each particle rsp. with the last one being the weight.

One important change: I had to remove the classmethod decorator over the parse_data_blocks method in the CorsikaFile class. 

The code passed the existing pytest modules. I have not added new modules to test for thin binary data files. However I've separately tested with thin data and things seems to be working fine so far. I might modify the test files to add new modules to check for the changes.

### CODE REVIEW

File: corsikaio/file.py [hunk 1]
```diff
@@ -33,24 +39,32 @@ class CorsikaFile:
...
         self._f = open_compressed(path)
-        self._block_iter = iter_blocks(self._f)
+        self._block_iter = iter_blocks(self._f, thinning=self.thinning)
 
         runh_bytes = next(self._block_iter)
         if not runh_bytes[:4] == b'RUNH':
             raise ValueError('File does not start with b"RUNH"')
 
-        self.run_header = parse_run_header(runh_bytes)[0]
+        if self.thinning == False:
```
Comment (maxnoe) [inline corsikaio/file.py:55]:
```suggestion
        if self.thinning is False:
```

File: corsikaio/file.py [hunk 2]
```diff
@@ -33,24 +39,32 @@ class CorsikaFile:
...
+            self.run_header = parse_run_header(runh_bytes)[0]
+        else:
+            self.run_header = parse_run_header_thin(runh_bytes)[0]
         self.version = round(float(self.run_header['version']), 4)
         self._run_end = None
 
     @property
     def run_end(self):
+        if self.thinning == False:
+            block_size = BLOCK_SIZE_BYTES
```
Comment (maxnoe) [inline corsikaio/file.py:65]: Could be assigned as `self.block_size` in `__init__` simplifying the logic here

File: corsikaio/subblocks/data.py [hunk 3]
```diff
@@ -12,6 +12,19 @@
 
 cherenkov_photons_dtype = build_dtype(cherenkov_photons_fields, itemsize=None)
 
+cherenkov_photons_fields_thin = [
```
Comment (maxnoe) [inline corsikaio/subblocks/data.py:15]: You could just reuse the fields and do:
```python
cherenkov_photons_fields_thin  = cherenkov_photons_fields + [Field(8, "thinning_level"), dtype="float32")]
```

File: corsikaio/subblocks/data.py [hunk 4]
```diff
@@ -37,3 +50,16 @@
 ]
 
 particle_data_dtype = build_dtype(particle_data_fields, itemsize=None)
+
+particle_data_fields_thin = [
```
Comment (maxnoe) [inline corsikaio/subblocks/data.py:54]: same here

File: corsikaio/subblocks/event_end.py [hunk 5]
```diff
@@ -39,10 +39,17 @@
 event_end_dtype_65 = build_dtype(event_end_fields_65)
 event_end_dtype_7x = build_dtype(event_end_fields_7x)
 
+event_end_thin_dtype_65 = build_dtype(event_end_fields_65, itemsize = 4 * 312)
```
Comment (maxnoe) [inline corsikaio/subblocks/event_end.py:42]: Use the constant you defined for this

File: tests/test_file.py [hunk 6]
```diff
@@ -8,19 +8,19 @@
 def test_version():
     from corsikaio import CorsikaFile
 
-    with CorsikaFile('tests/resources/mmcs65') as f:
+    with CorsikaFile('../tests/resources/mmcs65') as f:
```
Comment (maxnoe) [inline tests/test_file.py:11]: Please do not change the paths here.

Just run `pytest` from the base directory

**Comment by maxnoe** [comment]
Hi,

Thanks @oven8, this looks mostly good.

I think we mainly need a smallish test file with the thin option enabled to make sure this keeps working.

Some of the duplications can be removed and the test paths change needs to be reverted.

**Comment by oven8** [comment]
> Hi,
> 
> Thanks @oven8, this looks mostly good.
> 
> I think we mainly need a smallish test file with the thin option enabled to make sure this keeps working.
> 
> Some of the duplications can be removed and the test paths change needs to be reverted.

I'll add an update with the changes you recommended as well as a small test module.

File: corsikaio/subblocks/event_end.py (hunk 5 above)
Comment (oven8) [reply corsikaio/subblocks/event_end.py:42]: Do you mean to import BLOCK_SIZE_BYTES_THIN from ..constants and use that to define the itemsize? In that case should I also implement the same (BLOCK_SIZE_BYTES) for the itemsize in build_dtype method of dtypes.py?

**Comment by oven8** [comment]
I have added thin (Cherenkov) binary data file and modified some of the test modules. Let me know if more changes are required.

**Comment by maxnoe** [comment]
@oven8 This looks mostly good, thanks!

The one concern I have is that the particle test file at 20 MB is quite large for a test file. Could you maybe create a smaller one, the Cherenkov file is fine size-wise!

**Comment by oven8** [comment]
> @oven8 This looks mostly good, thanks!
> 
> The one concern I have is that the particle test file at 20 MB is quite large for a test file. Could you maybe create a smaller one, the Cherenkov file is fine size-wise!

I've swapped the thin test file with a smaller one (previously there were 200 events now 5). Let me know if this is okay.

**Comment by maxnoe** [comment]
@oven8 I am sorry this took so long, looks good now. Feel free to ping me more often next time.

I am going to squash the commits here to make sure the larger test files that were here in intermediate commits don't land in the project history.

**Review by maxnoe — APPROVED** [review]
Thanks a lot for this contribution!

**Comment by oven8** [comment]
Thanks for all the help! It was nice collaborating with you!

**Pull request merged**